### PR TITLE
Updated Parameter Input Validation.

### PR DESCRIPTION
**Pull request by relmota**
*Issue #, if available:*

*Description of changes:*
verified all parameter inputs in cfn templates and SSM document resource , added regex patterns where needed and updated existing ones as well as checking and updating numeric values.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: templates/ad-1.template [hunk 1]
```diff
@@ -278,20 +283,21 @@ Parameters:
     MinLength: '5'
     Type: String
   DomainDNSName:
-    AllowedPattern: '^[a-zA-Z0-9\-\.]+$'
+    AllowedPattern: '^[a-zA-Z0-9]+[a-zA-Z0-9\-]*\.[a-zA-Z0-9]+$'
     Default: example.com
     Description: Fully qualified domain name (FQDN) of the forest root domain e.g. example.com
-    MaxLength: '255'
-    MinLength: '2'
+    MaxLength: '25'
```
Comment (girvenj) [inline templates/ad-1.template:289]: Why 25?  That actual limit is 255

https://learn.microsoft.com/en-us/troubleshoot/windows-server/identity/naming-conventions-for-computer-domain-site-ou#dns-domain-names

File: templates/ad-1.template [hunk 2]
```diff
@@ -278,20 +283,21 @@ Parameters:
...
+    MinLength: '3'
```
Comment (girvenj) [inline templates/ad-1.template:290]: Why 3?  The actual min is 2.

https://learn.microsoft.com/en-us/troubleshoot/windows-server/identity/naming-conventions-for-computer-domain-site-ou#dns-domain-names

File: templates/ad-1.template [hunk 3]
```diff
@@ -438,44 +444,67 @@ Resources:
...
             description: 'AWS Secrets Parameter Name for the Active Directory Restore Mode Password'
             type: 'String'
+            maxChars: '2048'
+            minChars: '20'
           DomainDNSName:
+            allowedPattern: '^[a-zA-Z0-9]+[a-zA-Z0-9\-]*\.[a-zA-Z0-9]+$'
             default: 'example.com'
             description: 'Fully qualified domain name (FQDN) of the forest root domain e.g. example.com'
             type: 'String'
+            maxChars: '25'
```
Comment (girvenj) [inline templates/ad-1.template:499]: See Above

File: templates/ad-1.template [hunk 4]
```diff
@@ -438,44 +444,67 @@ Resources:
...
+            minChars: '3'
```
Comment (girvenj) [inline templates/ad-1.template:500]: See Above

File: templates/ad-1.template.yaml [hunk 5]
```diff
@@ -458,44 +463,68 @@ Resources:
...
             description: 'AWS Secrets Parameter Name for the Active Directory Restore Mode Password'
             type: 'String'
+            maxChars: '2048'
+            minChars: '20'
           DomainDNSName:
+            allowedPattern: '^[a-zA-Z0-9]+[a-zA-Z0-9\-]*\.[a-zA-Z0-9]+$'
             default: 'example.com'
             description: 'Fully qualified domain name (FQDN) of the forest root domain e.g. example.com'
             type: 'String'
+            maxChars: '25'
```
Comment (girvenj) [inline templates/ad-1.template.yaml:519]: See Above

File: templates/ad-1.template.yaml [hunk 6]
```diff
@@ -458,44 +463,68 @@ Resources:
...
+            minChars: '3'
```
Comment (girvenj) [inline templates/ad-1.template.yaml:520]: See Above

File: templates/ad-2.template [hunk 7]
```diff
@@ -141,28 +141,31 @@ Parameters:
...
+    MinValue: '1' 
+    MaxValue: '16384'
   DomainDNSName:
-    AllowedPattern: '^[a-zA-Z0-9\-\.]+$'
+    AllowedPattern: '^[a-zA-Z0-9]+[a-zA-Z0-9\-]*\.[a-zA-Z0-9]+$'
     Default: example.com
     Description: Fully qualified domain name (FQDN) of the domain you would like to join and promote to e.g. example.com
-    MaxLength: '255'
-    MinLength: '2'
+    MaxLength: '25'
```
Comment (girvenj) [inline templates/ad-2.template:157]: See Above

File: templates/ad-2.template [hunk 8]
```diff
@@ -141,28 +141,31 @@ Parameters:
...
+    MinLength: '3'
```
Comment (girvenj) [inline templates/ad-2.template:158]: See Above

File: templates/ad-2.template [hunk 9]
```diff
@@ -664,10 +697,10 @@ Resources:
             description: Fixed private IP for the second additional Active Directory Domain Controller located in subnet 2
             type: String
           DomainDNSName:
-            allowedPattern: '^[a-zA-Z0-9\-\.]+$'
+            allowedPattern: '^[a-zA-Z0-9]+[a-zA-Z0-9\-]*\.[a-zA-Z0-9]+$'
             description: Fully qualified domain name (FQDN) of the domain you would like to join and promote to e.g. example.com
-            maxChars: '255'
-            minChars: '2'
+            maxChars: '25'
```
Comment (girvenj) [inline templates/ad-2.template:702]: See Above

File: templates/ad-2.template [hunk 10]
```diff
@@ -664,10 +697,10 @@ Resources:
...
+            minChars: '3'
```
Comment (girvenj) [inline templates/ad-2.template:703]: See Above

File: templates/ad-2.template.yaml [hunk 11]
```diff
@@ -160,15 +160,17 @@ Parameters:
...
+    MinValue: '1' 
+    MaxValue: '16384'
   DomainDNSName:
-    AllowedPattern: '^[a-zA-Z0-9\-\.]+$'
+    AllowedPattern: '^[a-zA-Z0-9]+[a-zA-Z0-9\-]*\.[a-zA-Z0-9]+$'
     Default: example.com
     Description: Fully qualified domain name (FQDN) of the domain you would like to join and promote to e.g. example.com
-    MaxLength: '255'
-    MinLength: '2'
+    MaxLength: '25'
```
Comment (girvenj) [inline templates/ad-2.template.yaml:169]: See Above

File: templates/ad-2.template.yaml [hunk 12]
```diff
@@ -160,15 +160,17 @@ Parameters:
...
+    MinLength: '3'
```
Comment (girvenj) [inline templates/ad-2.template.yaml:170]: See Above

File: templates/ad-2.template.yaml [hunk 13]
```diff
@@ -291,17 +292,30 @@ Resources:
...
             description: NetBIOS name of the first Active Directory Domain Controller (up to 15 characters)
             type: String
+            maxChars: '15'
+            minChars: '1'  
           DomainDNSName:
+            allowedPattern: '^[a-zA-Z0-9]+[a-zA-Z0-9\-]*\.[a-zA-Z0-9]+$'
             description: Fully qualified domain name (FQDN) of the domain you would like to join and promote to e.g. example.com
+            default: 'example.com'
             type: String
+            maxChars: '25'
```
Comment (girvenj) [inline templates/ad-2.template.yaml:311]: See Above

File: templates/ad-2.template.yaml [hunk 14]
```diff
@@ -291,17 +292,30 @@ Resources:
...
+            minChars: '3'
```
Comment (girvenj) [inline templates/ad-2.template.yaml:312]: See Above

File: templates/ad-2.template.yaml [hunk 15]
```diff
@@ -684,10 +709,10 @@ Resources:
             description: Fixed private IP for the second additional Active Directory Domain Controller located in subnet 2
             type: String
           DomainDNSName:
-            allowedPattern: '^[a-zA-Z0-9\-\.]+$'
+            allowedPattern: '^[a-zA-Z0-9]+[a-zA-Z0-9\-]*\.[a-zA-Z0-9]+$'
             description: Fully qualified domain name (FQDN) of the domain you would like to join and promote to e.g. example.com
-            maxChars: '255'
-            minChars: '2'
+            maxChars: '25'
```
Comment (girvenj) [inline templates/ad-2.template.yaml:714]: See Above

File: templates/ad-2.template.yaml [hunk 16]
```diff
@@ -684,10 +709,10 @@ Resources:
...
+            minChars: '3'
```
Comment (girvenj) [inline templates/ad-2.template.yaml:715]: See Above

File: templates/ad-3.template [hunk 17]
```diff
@@ -185,28 +188,28 @@ Parameters:
...
     NoEcho: 'true'
     Type: String
   DomainDNSName:
-    AllowedPattern: '^[a-zA-Z0-9\-\.]+$'
+    AllowedPattern: '^[a-zA-Z0-9]+[a-zA-Z0-9\-]*\.[a-zA-Z0-9]+$'
     Default: example.com
     Description: Fully qualified domain name (FQDN) of the forest root domain e.g. example.com
-    MaxLength: '255'
-    MinLength: '2'
+    MaxLength: '25'
```
Comment (girvenj) [inline templates/ad-3.template:201]: See Above

File: templates/ad-3.template [hunk 18]
```diff
@@ -185,28 +188,28 @@ Parameters:
...
+    MinLength: '3'
```
Comment (girvenj) [inline templates/ad-3.template:202]: See Above

File: templates/ad-3.template.yaml [hunk 19]
```diff
@@ -197,28 +200,28 @@ Parameters:
...
     NoEcho: 'true'
     Type: String
   DomainDNSName:
-    AllowedPattern: '^[a-zA-Z0-9\-\.]+$'
+    AllowedPattern: '^[a-zA-Z0-9]+[a-zA-Z0-9\-]*\.[a-zA-Z0-9]+$'
     Default: example.com
     Description: Fully qualified domain name (FQDN) of the forest root domain e.g. example.com
-    MaxLength: '255'
-    MinLength: '2'
+    MaxLength: '25'
```
Comment (girvenj) [inline templates/ad-3.template.yaml:213]: See Above

File: templates/ad-3.template.yaml [hunk 20]
```diff
@@ -197,28 +200,28 @@ Parameters:
...
+    MinLength: '3'
```
Comment (girvenj) [inline templates/ad-3.template.yaml:214]: See Above

File: templates/ad-main-1.template.yaml [hunk 21]
```diff
@@ -285,40 +288,43 @@ Parameters:
...
     MinLength: '5'
     Type: String
   DomainDNSName:
-    AllowedPattern: '[a-zA-Z0-9\-]+\..+'
+    AllowedPattern: '^[a-zA-Z0-9]+[a-zA-Z0-9\-]*\.[a-zA-Z0-9]+$'
     Default: example.com
     Description: Fully qualified domain name (FQDN) of the forest root domain e.g. example.com
-    MaxLength: '255'
-    MinLength: '2'
+    MaxLength: '25'
```
Comment (girvenj) [inline templates/ad-main-1.template.yaml:316]: See Above

File: templates/ad-main-1.template.yaml [hunk 22]
```diff
@@ -285,40 +288,43 @@ Parameters:
...
+    MinLength: '3'
```
Comment (girvenj) [inline templates/ad-main-1.template.yaml:317]: See Above

File: templates/ad-main-2.template.yaml [hunk 23]
```diff
@@ -188,28 +188,30 @@ Parameters:
...
+    MaxValue: '16384'
     Type: Number
   DomainDNSName:
-    AllowedPattern: '[a-zA-Z0-9\-]+\..+'
+    AllowedPattern: '^[a-zA-Z0-9]+[a-zA-Z0-9\-]*\.[a-zA-Z0-9]+$'
     Default: example.com
     Description: Fully qualified domain name (FQDN) of the domain you would like to join and promote to e.g. example.com example.com
-    MaxLength: '255'
-    MinLength: '2'
+    MaxLength: '25'
```
Comment (girvenj) [inline templates/ad-main-2.template.yaml:198]: See Above

File: templates/ad-main-2.template.yaml [hunk 24]
```diff
@@ -188,28 +188,30 @@ Parameters:
...
+    MinLength: '3'
```
Comment (girvenj) [inline templates/ad-main-2.template.yaml:199]: See Above

File: templates/ad-main-3.template.yaml [hunk 25]
```diff
@@ -213,28 +216,28 @@ Parameters:
...
     NoEcho: 'true'
     Type: String
   DomainDNSName:
-    AllowedPattern: '[a-zA-Z0-9\-]+\..+'
+    AllowedPattern: '^[a-zA-Z0-9]+[a-zA-Z0-9\-]*\.[a-zA-Z0-9]+$'
     Default: example.com
     Description: Fully qualified domain name (FQDN) of the forest root domain e.g. example.com
-    MaxLength: '255'
-    MinLength: '2'
+    MaxLength: '25'
```
Comment (girvenj) [inline templates/ad-main-3.template.yaml:229]: See Above

File: templates/ad-main-3.template.yaml [hunk 26]
```diff
@@ -213,28 +216,28 @@ Parameters:
...
+    MinLength: '3'
```
Comment (girvenj) [inline templates/ad-main-3.template.yaml:230]: See Above

File: templates/mgmt-1.template.yaml [hunk 27]
```diff
@@ -103,21 +107,23 @@ Parameters:
     Description: IP of DNS server that can resolve domain (Must be accessible)
     Type: String
   DomainDNSName:
-    AllowedPattern: '^[a-zA-Z0-9\-\.]+$'
+    AllowedPattern: '^[a-zA-Z0-9]+[a-zA-Z0-9\-]*\.[a-zA-Z0-9]+$'
     Description: Fully qualified domain name (FQDN) of the forest root domain e.g. example.com
-    MaxLength: '255'
-    MinLength: '2'
+    MaxLength: '25'
```
Comment (girvenj) [inline templates/mgmt-1.template.yaml:112]: See Above

File: templates/mgmt-1.template.yaml [hunk 28]
```diff
@@ -103,21 +107,23 @@ Parameters:
...
+    MinLength: '3'
```
Comment (girvenj) [inline templates/mgmt-1.template.yaml:113]: See Above

**Review by girvenj — CHANGES_REQUESTED** [review]
I am not why this PR is setting the Doman name to such a short max value.  See here for proper name standards: https://learn.microsoft.com/en-us/troubleshoot/windows-server/identity/naming-conventions-for-computer-domain-site-ou#dns-domain-names

File: templates/ad-1.template (hunk 1 above)
Comment (relmota) [reply templates/ad-1.template:289]: That it correct , it should be 2-255 , will resolve that as well as the rest of them.

**Pull request closed without merging**